app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/OverScrollLayout.java

```diff
@@ -0,0 +1,171 @@
+package io.github.droidkaigi.confsched2017.view.customview;
+
+import android.animation.Animator;
+import android.content.Context;
+import android.databinding.BindingMethod;
+import android.databinding.BindingMethods;
+import android.graphics.PointF;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.AppBarLayout;
+import android.support.design.widget.CoordinatorLayout;
+import android.support.v4.widget.NestedScrollView;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.animation.AccelerateDecelerateInterpolator;
+
+import io.github.droidkaigi.confsched2017.R;
+
+
+@BindingMethods({
+        @BindingMethod(type = OverScrollLayout.class, attribute = "onOverScroll", method = "setOverScrollListener")
+})
+public class OverScrollLayout extends CoordinatorLayout {
+
+    private static final float OVER_SCROLL_THRESHOLD = 0.20f;
+
+    private static final int RESTORE_ANIM_DURATION = 100;
+
+    private static final float MINIMUM_OVER_SCROLL_SCALE = 0.99f;
+
+    private int overScrollThreshold;
+
+    private PointF originalLocation = new PointF();
```

Comment:
`Rect` is better, I think. Then, Rect#getHeight can be used instead of originalHeight.